lms/templates/courseware/courses.html

```diff
@@ -25,9 +26,11 @@
               credit_courses.append(x)
           else:
             if x.display_number_with_default.startswith(tuple(codes)):
-              credit_courses.append(x)
+              credit_courses_letter.append(x)
             else:
               non_credit_courses.append(x)
+        import re
+        credit_courses.sort(key=lambda x: float(re.sub(r'[^0-9\.]', '', x.display_number_with_default)))
```

Comment:
is display_number_with_default always there. for happen of it is None?

Reply:
I think it corresponds to the course number, and it is required:
![screen shot 2018-05-07 at 11 00 19 am](https://user-images.githubusercontent.com/7574259/39708848-483c293a-51e6-11e8-83c0-fa83751f1fc9.png)